hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/stats/SidecarInstrumentationImpl.java

```diff
@@ -0,0 +1,208 @@
+        final var acc = accumulators.get(type);
+        acc.computeDuration().addSample(duration.toNanos());
+    }
+
+    /**
+     * Add a sample of both serialized size and compute time for a sidecar transaction, for a
+     * particular kind of sidecar
+     */
+    @Override
+    public void addSample(
+            @NonNull final SidecarType type, long size, @NonNull final Duration duration) {
+        final var acc = accumulators.get(type);
+        acc.serializedSize().addSample(size);
+        acc.computeDuration().addSample(duration.toNanos());
+    }
+
+    /**
+     * Add the entire accumulated values of a SidecarInstrumentation to this one (which is a summary
+     * of multiple records
+     */
+    @SuppressWarnings("java:S2445")
+    // sync on method parameter _OK_ because adding samples goes only in _one_ direction ever (from
+    // more specific accumulator to more general accumulator)
+    @Override
+    public void addSamples(@NonNull final SidecarInstrumentation samples) {
+        if (samples instanceof SidecarInstrumentationImpl impl) {
+            synchronized (this) {
+                synchronized (samples) {
+                    for (final var e : SidecarType.values()) {
+                        final var thisAcc = accumulators.get(e);
+                        final var thatAcc = impl.accumulators.get(e);
+                        thisAcc.computeDuration().addSamples(thatAcc.computeDuration());
+                        thisAcc.serializedSize().addSamples(thatAcc.serializedSize());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Start timing a split. Use the returned record in a try-with-resources that encloses the
+     * computation to be timed.
+     */
+    @Override
+    public Closeable startTimer(@NonNull final SidecarType type) {
+        return new DurationTimer(this, type, clock.instant());
+    }
+
+    /**
+     * Measures a duration split, adding the elapsed duration to the given accumulator when
+     * `close()` is called (probably by a try-with-resources statement).
+     */
+    private record DurationTimer(
+            @NonNull SidecarInstrumentationImpl parent,
+            @NonNull SidecarType type,
+            @NonNull Instant start)
+            implements Closeable {
+        public void close() {
+            final var now = parent.clock().instant();
+            final var interval = Duration.between(start, now);
+            parent.accumulators().get(type).durationSplit().addSample(interval.toNanos());
+        }
+    }
+
+    /**
+     * Measure a duration split for a computation given by a `Runnable`, add that duration split as
+     * a sample.
+     */
+    @Override
+    public void captureDurationSplit(@NonNull final SidecarType type, @NonNull final Runnable fn) {
+        try (final var timer = startTimer(type)) {
+            fn.run();
+        } catch (IOException ex) { // because Closeable
+            throw new UncheckedIOException(ex);
```

Comment:
it can't actually be thrown - it's just there because  `Closeable` requires it.  I'll add a message as suggested but is there another way to do this (I mean "handle" an exception that can't be thrown that is just there because the interface demands it?)